src/librustc_mir_build/thir/pattern/const_to_pat.rs

```diff
@@ -274,12 +274,16 @@ impl<'a, 'tcx> ConstToPat<'a, 'tcx> {
                     self.behind_reference.set(old);
                     val
                 }
-                // Backwards compatibility hack. Don't take away the reference, since
-                // `PartialEq::eq` takes a reference, this makes the rest of the matching logic
-                // simpler.
+                // Backwards compatibility hack. Don't take away the reference, since we'll lower
```

Comment:
```suggestion
                // Backwards compatibility hack: support references to non-structural types.
                // Don't take away the reference, since we'll lower
```